Lib/pdb.py

```diff
@@ -414,16 +416,32 @@ def preloop(self):
                     self.message('display %s: %r  [old: %r]' %
                                  (expr, newvalue, oldvalue))
 
-    def interaction(self, frame, traceback):
+    def interaction(self, frame, tb_or_exc):
         # Restore the previous signal handler at the Pdb prompt.
+
+        _exceptions = []
+        if isinstance(tb_or_exc, BaseException):
+            tb_or_exc, exception = tb_or_exc.__traceback__, tb_or_exc
+            current = exception
+            while current:
+                _exceptions.append(current)
+                if current.__cause__ is not None:
+                    current = current.__cause__
+                elif current.__context__ is not None and not current.__suppress_context__:
+                    current = current.__context__
+                else:
+                    current = None
+        self._chained_exceptions = _exceptions[::-1]
```

Comment:
```suggestion
        self._chained_exceptions = tuple(reversed(_exceptions))
```

Reply:
I'm curious, why `tuple(reversed(...))` and not `[::-1]`. tuple to make immutable is better, but I though `[::-1]` was more efficient on list.

Reply:
actually scratch that, wouldn't

```
_exceptions.reverse()
tuple(_exceptions)
```

be even faster ?

Reply:
My motivation here was readability rather than efficiency. It's all linear anyway.